src/improvements/doc/TEMPLATE_ARCHITECTURE.md

```diff
@@ -0,0 +1,156 @@
+# Template Architecture Guide
+
+This guide explains the architecture of the template system, including configuration structure, file organization, and validation requirements.
+
+## Directory Structure
+
+Templates are organized in the following structure:
+
+```
+src/improvements/
+├── template/          # Template Solidity contracts
+├── doc/               # Documentation
+└── tasks/             # Network-specific tasks
+     ├── eth/          # Ethereum mainnet tasks
+     ├── sep/          # Sepolia testnet tasks
+     ├── oeth/         # Optimism Ethereum tasks
+     └── opsep/        # Optimism Sepolia tasks
+```
+
+## Template Configuration Structure
+
+### 1. Template Location
+
+Templates are Solidity contracts located in the `src/improvements/template/` directory. Each template is designed for a specific type of task operation.
+
+### 2. Configuration File Architecture
+
+Each task requires a `config.toml` file with two main components:
+
+1. L2 Chain Configuration
+2. Template Selection
+
+#### L2 Chain Configuration
+
+Every config file must specify the L2 chains that the task will interact with:
```

Comment:
Right now, I think it's true that all chains that are added into this array _must_ have the same proxy admin owner multisig. Otherwise this reverts.